collector/setting.go

```diff
@@ -0,0 +1,96 @@
+package collector
+
+import (
+	rancher "github.com/rancher/types/client/management/v3"
+	log "github.com/sirupsen/logrus"
+)
+
+func GetSetting(client *rancher.Client, key string) (string, bool) {
+	return GetSettingByCollection(GetSettingCollection(client), key)
+}
+
+func GetSettingCollection(client *rancher.Client) *rancher.SettingCollection {
+	opts := NonRemoved()
+	opts.Filters["all"] = "true"
+
+	settings, err := client.Setting.List(&opts)
+	if err != nil {
+		log.Errorf("GetSettingsCollection: Error: %s", err)
+		return nil
+	}
+
+	if settings == nil || settings.Type != "collection" || len(settings.Data) == 0 {
+		log.Debugf("Settings collection is empty")
+		return nil
+	}
+
+	return settings
+}
+
+func GetAllSettingData(client *rancher.Client) []rancher.Setting {
+	setList := GetSettingCollection(client)
+	if setList == nil {
+		return nil
+	}
+
+	setData := setList.Data
+	// Paginating if needed
+	setList, err := setList.Next()
+	for page := 2; setList != nil && err == nil; setList, err = setList.Next() {
+		log.Debug(" Paginating Settings page ", page)
+		setData = append(setData, setList.Data...)
+		page++
+	}
+	if err != nil {
+		log.Errorf(" Failed to paginate Settings err=%s", err)
+		return nil
+	}
+
+	return setData
+}
+
+func GetSettingByCollection(settings *rancher.SettingCollection, key string) (string, bool) {
+	return GetSettingByData(settings.Data, key)
```

Comment:
Is there any point to having 2 methods here ? Feels like the two can be combined.

Reply:
You are right, cleaned up `GetSettingByCollection` function